exercises/concept/elyses-analytic-enchantments/enchantments.spec.js

```diff
@@ -9,103 +9,55 @@ import {
-      });
+  describe('stack includes card', () => {
+    test('when there is a card found', () => {
+      expect(doesStackIncludeCard([1, 2, 3], 1)).toBe(true);
+    });
+    test('when a card is not found', () => {
+      expect(doesStackIncludeCard([1, 2, 3], 4)).toBe(false);
     });
   });
 
-  describe('isEachCardEven', () => {
-    /** @type {TestAllMatrix<boolean>} */
-    const isEachCardEvenTestCases = [
-      [[1], false],
-      [[2, 5], false],
-      [[2, 4, 8, 6], true],
-    ];
-
-    isEachCardEvenTestCases.forEach(([array, expected]) => {
-      test(`isEachCardEven([${array}])`, () => {
-        expect(isEachCardEven(array)).toStrictEqual(expected);
-      });
+  describe('is each card even', () => {
+    test('when all cards are even', () => {
+      expect(isEachCardEven([2, 4, 6])).toBe(true);
+    });
+    test('when any card is odd', () => {
+      expect(isEachCardEven([2, 5, 6])).toBe(false);
     });
   });
 
-  describe('doesStackIncludeOddCard', () => {
-    /** @type {TestAllMatrix<boolean>} */
-    const doesStackIncludesOddCardTestCases = [
-      [[2, 4, 6], false],
-      [[2, 5], true],
-      [[1, 3, 5, 7], true],
-    ];
-
-    doesStackIncludesOddCardTestCases.forEach(([array, expected]) => {
-      test(`doesStackIncludeOddCard([${array}])`, () => {
-        expect(doesStackIncludeOddCard(array)).toStrictEqual(expected);
-      });
+  describe('does stack include odd card', () => {
+    test('should be true if odd number card is found', () => {
+      expect(doesStackIncludeOddCard([2, 5])).toBe(true);
+    });
+    test('should be false if no odd number card is found', () => {
+      expect(doesStackIncludeOddCard([2, 4, 6])).toBe(false);
     });
   });
 
   describe('getFirstOddCard', () => {
-    /** @type {TestAllMatrix<number | undefined>} */
-    const getFirstOddCardTestCases = [
-      [[2, 4, 1, 3], 1],
-      [[1, 2], 1],
-      [[4, 2, 6], undefined],
-    ];
-
-    getFirstOddCardTestCases.forEach(([array, expected]) => {
-      test(`getFirstOddCard([${array}])`, () => {
-        expect(getFirstOddCard(array)).toStrictEqual(expected);
-      });
+    test('should return the first odd card found', () => {
+      expect(getFirstOddCard([2, 4, 1, 3])).toBe(1);
+    });
+    test('should return undefined if odd card is not found', () => {
+      expect(getFirstOddCard([4, 2, 6])).toBe(undefined);
     });
   });
 
   describe('getFirstEvenCardPosition', () => {
-    /** @type {TestAllMatrix<number>} */
-    const getFirstEvenCardPositionTestCases = [
-      [[2, 4, 1, 3], 0],
-      [[1, 2], 1],
-      [[1, 3, 5], -1],
-    ];
-
-    getFirstEvenCardPositionTestCases.forEach(([array, expected]) => {
-      test(`getFirstEvenCardPosition([${array}])`, () => {
-        expect(getFirstEvenCardPosition(array)).toStrictEqual(expected);
-      });
+    test('should return position of first even card', () => {
+      expect(getFirstEvenCardPosition([2, 4, 1, 3])).toBe(0);
```

Comment:
The "-1" case is not tested.

Also if you feel like adding another assertion for a similar thing in any of the tests, you can also use multiple `expect` statements in one `test` block.

Reply:
> The "-1" case is not tested.

This is not addressed yet.

Reply:
:point_up: The "-1" test case is still missing.